iothub/device/src/InternalClient.cs

```diff
@@ -1895,7 +1895,6 @@ public void Dispose()
             _fileUploadHttpTransportHandler?.Dispose();
             _deviceReceiveMessageSemaphore?.Dispose();
             _twinDesiredPropertySemaphore?.Dispose();
-            IotHubConnectionString?.TokenRefresher?.Dispose();
```

Comment:
This is a breaking change. If we don't dispose it, and a customer had been relying on us to do that, now they have a leak.

You need to follow the breaking change process here.

Reply:
The user would need to explicitly dispose it only for scenarios where they supply the sdk with an `AuthenticationWithTokenRefresh` implementation.

Reply:
It is a change of behavior.